src/traits.rs

```diff
@@ -305,4 +312,62 @@ pub mod test_helpers {
             panic!("DKG should have failed")
         }
     }
+
+    /// Check that bad polynomial commitments are properly caught as errors during DKG
+    pub fn bad_polynomial_commitment<Signer: super::Signer>() {
+        let num_keys: u32 = 10;
+        let num_signers: u32 = 4;
+        let threshold: u32 = 7;
+        let signer_ids: Vec<Vec<u32>> = vec![vec![1, 2, 3, 4], vec![5, 6, 7], vec![8, 9], vec![10]];
+        let mut rng = create_rng();
+        let mut signers: Vec<Signer> = signer_ids
+            .iter()
+            .enumerate()
+            .map(|(id, ids)| {
+                Signer::new(
+                    id.try_into().unwrap(),
+                    ids,
+                    num_signers,
+                    num_keys,
+                    threshold,
+                    &mut rng,
+                )
+            })
+            .collect();
+
+        let bad_party_id = 2u32;
+        let public_shares: HashMap<u32, PolyCommitment> = signers
```

Comment:
Can we add a comment that says something like
```rust
// The code that follows is essentially the same code that we have
// in the `dkg` helper function above, except we've corrupted the
// schnorr proof so that we can test verification would fail at
// the end.
```

Reply:
Done